org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/rule/RuleDescriptionPanel.java

```diff
@@ -49,76 +54,151 @@
+    var currentHTML = html;
+    var matcherStart = Pattern.compile("<pre[^>]*>").matcher(currentHTML);
+    var matcherEnd = Pattern.compile("</pre>").matcher(currentHTML);
+
+    while (matcherStart.find() && matcherEnd.find()) {
+      var front = currentHTML.substring(0, matcherStart.start()).trim();
+      if (!front.isEmpty() && !front.isBlank()) {
+        var frontFragment = new SonarLintWebView(parent, useEditorFontSize);
+        var gridData = new GridData();
+        gridData.horizontalAlignment = SWT.FILL;
+        gridData.grabExcessHorizontalSpace = true;
+        frontFragment.setLayoutData(gridData);
+
+        frontFragment.setHtmlBody(front);
+      }
+
+      var middle = currentHTML.substring(matcherStart.end(), matcherEnd.start()).trim();
+      if (!middle.isEmpty() && !middle.isBlank()) {
+        createSourceViewer(StringUtils.xmlDecode(middle), parent, languageKey);
+      }
+
+      currentHTML = currentHTML.substring(matcherEnd.end(), currentHTML.length()).trim();
+      matcherStart = Pattern.compile("<pre[^>]*>").matcher(currentHTML);
+      matcherEnd = Pattern.compile("</pre>").matcher(currentHTML);
+    }
+
+    if (!currentHTML.isEmpty() && !currentHTML.isBlank()) {
+      var endFragment = new SonarLintWebView(parent, useEditorFontSize);
+      var gridData = new GridData();
+      gridData.horizontalAlignment = SWT.FILL;
+      gridData.grabExcessHorizontalSpace = true;
+      endFragment.setLayoutData(gridData);
+
+      endFragment.setHtmlBody(currentHTML);
+    }
+  }
+
+  private static void createSourceViewer(String html, Composite parent, String languageKey) {
+    // Configure the syntax highlighting based on the rule language key and if a configuration and document partitioner
+    // is provided by any plug-in via the extension mechanism.
+    // INFO: Configuration must extend of org.eclipse.jface.text.source.SourceViewerConfiguration
+    // INFO: Document partitioner must implement org.eclipse.jface.text.IDocumentPartitioner
+    var configurationProviders = SonarLintExtensionTracker.getInstance().getSyntaxHighlightingProvider();
+    SourceViewerConfiguration sourceViewerConfigurationNullable = null;
+    for (var configurationProvider : configurationProviders) {
+      var sourceViewerConfigurationOptional = configurationProvider.sourceViewerConfiguration(languageKey);
+      if (sourceViewerConfigurationOptional.isPresent()) {
+        sourceViewerConfigurationNullable = sourceViewerConfigurationOptional.get();
+        break;
+      }
+    }
+
+    IDocumentPartitioner documentPartitionerNullable = null;
+    for (var configurationProvider : configurationProviders) {
+      var documentPartitionerOptional = configurationProvider.documentPartitioner(languageKey);
+      if (documentPartitionerOptional.isPresent()) {
+        documentPartitionerNullable = documentPartitionerOptional.get();
+        break;
+      }
+    }
+
+    var snippetElement = new SourceViewer(parent, null, SWT.BORDER);
```

Comment:
I think we should remove the border at some point :)

Reply:
Yeah, agree. Was only done for seeing the `SourceViewer` boundaries locally and when they change.

Reply:
In fact, I realized there is a border in the Eclipse formater settings, so in the end maybe we could keep it.